packages/scaffold/src/modal/w3m-network-button/index.tsx

```diff
@@ -33,15 +42,20 @@ export function NetworkButton({ disabled, style }: NetworkButtonProps) {
   return (
     <ThemeProvider themeMode={themeMode} themeVariables={themeVariables}>
       <NetworkButtonUI
-        imageSrc={AssetUtil.getNetworkImage(caipNetwork)}
+        imageSrc={AssetUtil.getNetworkImage(NetworkController.state.caipNetwork)}
         imageHeaders={ApiController._getApiHeaders()}
         disabled={disabled || loading}
         style={style}
         onPress={onNetworkPress}
         loading={loading}
         testID="network-button"
       >
-        {caipNetwork?.name ?? (isConnected ? 'Unknown Network' : 'Select Network')}
+        {NetworkController.state.caipNetwork?.name ??
```

Comment:
this one can be `caipNetwork?.name` but change this condition. Even if there's a caip network name, if it's not supported we need to show "Switch Network"

Reply:
1. If the caip network is supported, show the name of the network
2. If the caip network is not supported, put "Switch network"
3. If the caip network is supported but we don't have the name, put Unknown Network (not sure if it will happen)
4. If there's no caip network, put Select Network